### DISCUSSION TITLE
멘토링 과제 (2021.10.21~2021.11.01)

### DISCUSSION THREAD

**Discussion by hye-jj** [kikiru328/Bone_Detection#41] [Assignment]
1회차 점검

1. 소스코드 리뷰(OpenCV)
 —> 안돌아 가는 것 있음

2. OpenCV 

3. 과제의 범위를 어디까지?

 —> **모델의 정확도를 올리는 방향으로**

4. 골연령 예측하는 방법 찾아보기
---------------------------------------------------------------------------------------

2회차 점검

1. 이미지 전처리 실시
—> 공유한 파일 참고
—> **ROI 추출 방법** 생각 -> OpenCV, AI

2. 골연령 인식 인공신경망 알고리즘 검색
—> **TJNet**

3. 예제 실행시켜보기

---------------------------------------------------------------------------------------

3회차 점검

1. 뼈 나이 보는 방법 두 가지 정리

2. ROI 추출 방법 더 보고 결정

3. 코드 탐색 더 해보세요.

---------------------------------------------------------------------------------------

4회차 점검

1. 뼈 나이 보는 방법 두 가지 정리
—> 잘 정리 했음

2. ROI 추출 방법 더 보고 결정
—> YOLO를 사용

3. 코드 탐색 더 해보세요.
Q&A
1. blur효과를 줬다가 다시 강조, 쓰레스홀드를 변경
2. 많을수록 좋음

다음시간...
성별 연령별로 YOLO이용 학습 ROI를
월까지 7개 바운딩박스, 라벨링

---------------------------------------------------------------------------------------

**Comment by hye-jj** [comment]
[ 21.10.22 15 : 40 ] 
1. 우선 승혜가 masking까지한 전처리 코드를 쭉 읽어가면서 한번 이상 해보고, 익숙해보는 것을 원칙으로 합시다.

2. 코드 내 더욱더 발전할 수 있는 가능성이 있는 다른 부분들이 있으면 공부해보고 공유 부탁드리겠습니다. (filtering -> 유빈님께서 새로운 방안을 찾아보셨습니다 : 확인 후 알려주세요) **잡음제거 -- blur**

3. BigROI를 짜르기 위해선 손목을 자르는 방법을 우선적으로 각자 공부해보고, 손가락을 자르기 위해서는 convex hull을 사용해야하는데 convex hull은 좀 어렵다고 합니다. 그래서 우선적으로는 convex hull 알고리즘을 최선으로 공부해보고, 이해가 어렵거나 어렵다면 손목 BIG ROI 추출을 먼저 해봅시다. 그리고 나중에 convex hull을 성공한 다른 팀원께서 나머지 팀원께 공유를 하고 공부해 보는 걸로 합시다

4. BIG ROI가 추출됐을때 아까 정했던 폴더에 정해진 파일명으로 나눠졌으면 다음 멘토링때 진행방안에 대해서 설명하면 될 것 같습니다

**Reply by hye-jj** [reply]
유빈님 1 - 240 / 혜정님 241- 500
광훈 여)501-573 / 남) 001-200
승혜님 201-450 / 종현님 451-665

**Reply by hye-jj** [reply]
사이즈 이진화 윤곽선 컨투어 -> 회전

**Reply by yibnn** [reply]
2번에 대한거 새로 만들었습니다 "잡음제거 -- blur" 확인해주세요

> https://github.com/kikiru328/Bone_Detection/discussions/43

**Comment by yibnn** [comment]
## 전처리 과정 ##
원본 -> 마스크생성 -> 마스크기준으로 원본에서 제거 -> 손목 회전 -> 손목 추출
-> 마스크기준으로 원본에서 제거된 사진 -> 각 손가락 엄지/중지/소지 손가락 ROI추출 (convex hull 사용)

**Reply by hye-jj** [reply]
확인했습니다! 감사합니다~

**Comment by hye-jj** [comment]
### #45 에 꼭 해야할일!! (20.10.22 회의 결과)
그럼 주말까지 코드 브레이킹을 할수있는 한 많이 해보는 것! 본인이 한 공부 DISCUSSION, preprocess code break 에 꼭 올리기!!!

**Comment by hye-jj** [comment]
## (20.10.25 회의 결과)

- 각자 전처리 코드 공부하고 정리한거 올리기.
- **회전, ROI추출이 중요**
- 필터 **bilateralFilter** 적용도 좋은 것을 확인함. 
- 승혜님 자료 참고해서 뼈나이 보는 방법 2가지 학습 및 정리.

**Reply by kikiru328** [reply]
감사합니다 선생님

**Reply by SHEYCHAE** [reply]
혜정이언니 최 고

**Comment by hye-jj** [comment]
## (20.10.29 회의 결과)

- 각자가 맡은 범위 바운딩박스치고 라벨달기!!!!

(작업과정 및 분할참고)
YOLO (YOU ONLY LIVE ONCE?) #54

(yolo 학습 라벨처리를 위한)
ROI를 위한 labelImg 사용방법 #55
